doc/flame/components.md

```diff
@@ -1,6 +1,71 @@
 # Components
 
-![Component tree](../images/component_tree.png)
+
+  ``` {mermaid}
+%%{init: { 'theme': 'dark' } }%%
```

Comment:
The mermaid diagrams should preferably be defined in their own files that then are just included in this file, so that it is easier to read the text of the raw markdown.

Reply:
Also the backticks should start at the beginning of the line or the markdown linter will complain. :)

Reply:
![Screen Shot 2022-10-02 at 4 01 57 PM](https://user-images.githubusercontent.com/36990991/193449630-7b6e46aa-3050-4300-83d8-dbc0f97da5b7.png)
![Screen Shot 2022-10-02 at 4 01 47 PM](https://user-images.githubusercontent.com/36990991/193449633-e0d93cc9-d5e6-4eba-80d8-30e7c7cdd4ed.png)
![Screen Shot 2022-10-02 at 4 01 39 PM](https://user-images.githubusercontent.com/36990991/193449634-0c3072c2-1f00-45ff-b1a8-1c4b22d57aa3.png)

Reply:
Nice! They look great!

Reply:
If you install `markdownlint-cli` (`npm install -g markdownlint-cli`) it can fix most, if not all, of the markdown warnings that you are getting from the pipeline by running `melos markdown-fix`.
To just see the warnings locally you can run `melos markdown-check`.
The current warnings are these:
```
 doc/flame/components.md:1 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "# Components"]
doc/flame/components.md:2 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```{include} diagrams/componen..."]
doc/flame/components.md:7 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 2; Actual: 1; Above] [Context: "## Component"]
doc/flame/diagrams/component_life_cycle.md:43:3 MD047/single-trailing-newline Files should end with a single newline character
doc/flame/diagrams/component.md:40:81 MD013/line-length Line length [Expected: 80; Actual: 87]
doc/flame/diagrams/component.md:60:4 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
doc/flame/diagrams/component.md:60 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
doc/flame/diagrams/component.md:61 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
doc/flame/diagrams/low_level_game_api.md:46:4 MD047/single-trailing-newline Files should end with a single newline character
```